### ISSUE TITLE
Inherited Objects are not serialized correctly

### ISSUE THREAD

**Issue by VasTheGreek** [DigDes/SoapCore#378]
This issue has been tested using the latest 1.0.0-alpha.3 release but it has been an issue with alpha2 as well. 

In my attached solution I have a test project and a SOAP service targetting in with Framework 4. I also have a newer implementation of a sample SOAP service in .net Core 3.1. The service reference has been obtained from the Framework 4 service and I do not wish to update the reference but only the port number. The service Data Contracts contain inherited classes. All that the service does is to return the submitted object.

Steps to reproduce:
- Framework service runs on port 8733 and Core Service on port 8734, which needs to be changed in the test project app.config. (In the committed solution it is set to point to port 8734).
- Start up the .Net Core implementation.
- Run the unit test and you will see that the properties on the object returned have their type default values and not the values that were set when submitted.

If you need any additional info please let me know.

[WebApplication2.zip](https://github.com/DigDes/SoapCore/files/3955300/WebApplication2.zip)

**Comment by kotovaleksandr** [comment]
@VasTheGreek Its not alpha3 issue, with alpha2 same behavior. Problem in data contracts namespaces. In generated client code all three Requests classed have namespace `http://schemas.datacontract.org/2004/07/WebApplication`, but on server side model some classes have namespace in declaration, some not.
If you add namespace declaration in all three models:
```csharp
[DataContract(Name = "GrandParentRequest", Namespace = "http://schemas.datacontract.org/2004/07/WebApplication")]
//[DataContract]
public class GrandParentRequest
```

```csharp
//[DataContract]
[DataContract(Name = "ParentRequest", Namespace = "http://schemas.datacontract.org/2004/07/WebApplication")]
public class ParentRequest : GrandParentRequest
```

```csharp
//[DataContract]
[DataContract(Name = "GrandParentRequest", Namespace = "http://schemas.datacontract.org/2004/07/WebApplication")]
public class UserRequest : ParentRequest
```

Test passed:
![image](https://user-images.githubusercontent.com/2936914/70709219-73920e00-1ced-11ea-9919-6ec785dc2028.png)

**Comment by VasTheGreek** [comment]
This has worked on demo solution that I uploaded. I think you got a typo on the DataContract for the "UserRequest" above in case you wish to update your answer. I will try on our main app too and will close the issue.

**Comment by kotovaleksandr** [comment]
Yes, its error, should be:
```csharp
[DataContract(Name = "UserRequest", Namespace = "http://schemas.datacontract.org/2004/07/WebApplication")]
public class UserRequest : ParentRequest
```

**Comment by VasTheGreek** [comment]
I got an additional question regarding the library's backward compatibility. From .Net 3.5 SP1 the [DataContract] attribute is optional (with some disadvantages), however the serialization of the object just works. 

I noticed that whether the [DataContract] attribute is used or not in the .Net Framework, implementation of the service the WSDL produced is identical in both cases. The client (tests) itself consumes the service and works correctly.

However when I change my client code to use the Core implementation of the service using SOAP Core (without updating the reference but only the endpoint), if the reference is created with the [DataContract] attribute present the test passes, but if it is not present the test fails. I don't understand how that is possible as the WSDL in both cases (with and without DataContract) is identical but also why does the SOAPCore library exclusively expects the [DataContract] attribute present?
